docs/fsharp/language-reference/sequences.md

```diff
@@ -1,50 +1,75 @@
 
 ## Sequence Expressions
 
 A *sequence expression* is an expression that evaluates to a sequence. Sequence expressions can take a number of forms. The simplest form specifies a range. For example, `seq { 1 .. 5 }` creates a sequence that contains five elements, including the endpoints 1 and 5. You can also specify an increment (or decrement) between two double periods. For example, the following code creates the sequence of multiples of 10.
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet1502.fs)]
 
-Sequence expressions are made up of F# expressions that produce values of the sequence. They can use the `yield` keyword to produce values that become part of the sequence.
-
-Following is an example.
+Sequence expressions are made up of F# expressions that produce values of the sequence. You can also generate values programmatically:
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet1503.fs)]
 
-You can use the `->` operator instead of `yield`, in which case you can omit the `do` keyword, as shown in the following example.
+The previous sample uses the `->` operator, which allows you to specify an expression whose value will become a part of the sequence. You can only use `->` if every part of the code that follows it returns a value.
+
+Alternatively, you can specify the `do` keyword, with an optional `yield` that follows:
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet1504.fs)]
 
-The following code generates a list of coordinate pairs along with an index into an array that represents the grid.
+The following code generates a list of coordinate pairs along with an index into an array that represents the grid. Note that the first `for` expression requires a `do` to be specified.
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet1505.fs)]
 
 An `if` expression used in a sequence is a filter. For example, to generate a sequence of only prime numbers, assuming that you have a function `isprime` of type `int -> bool`, construct the sequence as follows.
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet1506.fs)]
 
-When you use `yield` or `->` in an iteration, each iteration is expected to generate a single element of the sequence. If each iteration produces a sequence of elements, use `yield!`. In that case, the elements generated on each iteration are concatenated to produce the final sequence.
+As mentioned previously, `do` is required here because there is no `else` branch that goes with the `if`. If you try to use `->`, you will get an error saying that not all branches return a value.
```

Comment:
```suggestion
As mentioned previously, `do` is required here because there is no `else` branch that goes with the `if`. If you try to use `->`, you'll get an error saying that not all branches return a value.
```